MHCollapsibleViewManager.m

```diff
@@ -0,0 +1,228 @@
+//
+//  MHCollapsibleViewManager.m
+//  This is a manager of a uiview and view controller
+//
+//  Created by Lizzy Randall on 5/6/15.
+//  Copyright (c) 2015 Lizzy Randall. All rights reserved.
+//
+
+#import "MHCollapsibleViewManager.h"
+#import "MHFilterLabel.h"
+
+@implementation MHCollapsibleViewManager
+
+//@synthesize tableView;
+//@synthesize viewController;
+@synthesize rowAnimation;
+
+@synthesize data;
+@synthesize cellStyles;
+@synthesize headerTitle;
+
+@synthesize expanded;
+@synthesize numOfRows;
+
+//Initialize with animation, title and tableView
+- (id)initalizeManager:(UITableViewRowAnimation)animation title:(NSString*) title tableView:(UITableView*)tableView
+{
+    self.headerTitle = title;
+    self.rowAnimation = animation;
+    //initialized first and then can be overwritten by controller
+    //tableView is needed to create cells to store types
+    [self setCellProperties:nil header:nil clickedCell:nil normalCell:nil tableView:tableView];
+    return self;
+}
+
+//called after Initializing Manager
+- (void) initializeData:(NSMutableArray*)dataFromController
+{
+    self.data = [[NSMutableArray alloc] init];
+    for(int i=0; i < [dataFromController count] ; i++)
+    {
+        [data addObject:[MHFilterLabel constructLabel:[dataFromController objectAtIndex:i] checked:false]];
+    }
+    self.numOfRows = (NSInteger)[data count];
+}
+
+//set the cell provided with same settings as corresponding array type
+- (UITableViewCell*)setCellSettings:(UITableViewCell*)cell type:(NSString*)type
+{
+    UITableViewCell *tempCell = nil;
+    int index = 3; //normall cell
+    if([type isEqual: @"ClickedHeader"])
+    {
+        index = 0;
+    }
+    else if([type isEqual: @"NormalHeader"])
+    {
+        index = 1;
+    }
+    else if([type isEqual: @"ClickedCell"])
+    {
+        index = 2;
+    }
+    
+    tempCell = [cellStyles objectAtIndex:index];
```

Comment:
objc has a feature called literals https://github.com/michaelharro/Code-Style-Guidelines/blob/master/Languages/objective-c.md#literals
I loved them when I found them. It will make you life a lot easier if you use them.
This line would become
tempCell = cellStyles[index];